apps/docs/docs/state/selections/selections.mdx

```diff
@@ -28,16 +27,278 @@ npm install --save @rx-angular/state
+
+## smosh
+
+## distinctUntilSomeChanges
+
+# Advanced derivation architecture
+
+**The problem**
+
+We have the following state sources to manage:
+
+- the list of products received form global state - `Product[]`
+- the title of the list including it's number of children computen in the component class - `string`
+- the sort direction triggered over a UI element click - `boolean`
+
+A setup of the compoents class based on `RxState` could look like this:
+
+```typescript
+@Component({
+  selector: 'app-problem',
+  template: `
+    <ng-container *rxLet="viewModel$; let vm">
+      <h1>{{vm.title}} - {{vm.sortDirection}}</h1>
+      <ul>
+        <li *ngFor="let item of vm.sortedList">{{item}}</li>
+      </ul>
+    </ng-container>
+    `,
+  providers: [RxState],
+})
+export class ProblemComponent {
+
+  viewModel$: Observable<ViewModel>; // ???
+
+  constructor(private globalState: GlobalState, private state: RxState<Model>) {
+    this.state.connect('title', this.globalState.title$);
+    this.state.connect('products', this.globalState.products$);
+  }
+
+  toggleSort() {
+    this.state.set('sort', ({sort}) => !sort))
+  }
+}
+
+```
+
+In a components template we want to render the the UI for the above explained view model `SelectionScreen1`.
+
+```typescript
+interface SelectionScreen1 {
+  title: string;
+  sortDirection: 'asc' | 'desc' | 'none';
+  sortedList: Array<{ id: number }>;
+}
+```
+
+A common implementations looks like this:
+
+```typescript
+// template removed for brevity
+export class ProblemComponent {
+  private sortedList$ = this.state.select(
+    selectSlice(['sortDirection', 'list']),
+    map(() => {
+      // sort `list` by `sortDirection` to `sortedList` here
+      return sortedList;
+    })
+  );
+
+  viewModel$ = this.state.select(
+    selectSlice(['title', 'sortedList', 'sortDirection'])
+  );
+
+  //                                                    ❌ BAD: modle viewmodel mix up 👇
+  constructor(
+    private globalState: GlobalState,
+    private state: RxState<Model & Pick<ViewModel, 'sortedList'>>
+  ) {
+    // ...
+
+    // ❌ BAD: store derived state 👇
+    this.state.connect('sortedList', this.sortedList$);
+  }
+
+  // ...
+}
+```
+
+![Selections (6)](https://user-images.githubusercontent.com/10064416/152422999-db8260f0-69e1-4d99-b6ac-b2b1d043b4b7.png)
 
-- [Selections](https://rx-angular.io/docs/state/selections)
+By removing the sorted list form the state and moving it into the selection
+we can clean up the state's typing and have a nice separation of which data is owned by the component (model) and which data is owned by the template (view model)
 
-## Motivation
+```typescript
+// template removed for brevity
+export class ProblemComponent {
+  private sortedSlice$ = this.state.select(
+    selectSlice(['sortDirection', 'list']),
+    map(({ list, sortDirection }) => {
+      // sort `list` by `sortDirection` to `sortedList` here
+      return { sortDirection, sortedList };
+    })
+  );
```

Comment:
Minor suggestion, we could directly write:

```suggestion
  private sortedSlice$ = this.state.select(
    ['sortDirection', 'list'],
    ({ list, sortDirection }) => {
      // sort `list` by `sortDirection` to `sortedList` here
      return { sortDirection, sortedList };
    }
  );
```

Reply:
I'm imho this is not working. Is it a new api suggestion?

Reply:
yes it's part of the API but it's not documented